File: backend/app/schemas/analytics.py
"""Analytics-related Pydantic schemas."""

from pydantic import BaseModel


class AnalyticsSummary(BaseModel):
    """Schema for analytics summary response."""
    
    total_tasks: int
    completed_tasks: int
    pending_tasks: int
    in_progress_tasks: int
    overdue_tasks: int
    completed_this_week: int
    high_priority_pending: int
    completion_rate: float

File: frontend/src/services/analytics.ts
import api from './api';
import type { AnalyticsSummary } from '@/types';

export const analyticsService = {
  async getSummary(): Promise<AnalyticsSummary> {
    const response = await api.get<AnalyticsSummary>('/analytics/summary');
    return response.data;
  },
};

export default analyticsService;

